http/src/main/resources/init-resources/startup.sh

```diff
@@ -243,16 +247,9 @@ MEM_LIMIT=${MEM_LIMIT}
 SHM_SIZE=${SHM_SIZE}
 END
 
-        COMPLETE_JUPYTER_DOCKER_COMPOSE="-f $JUPYTER_DOCKER_COMPOSE"
-        if [ "${GPU_ENABLED}" == "true" ] ; then
-          GPU_DOCKER_COMPOSE=$(ls ${DOCKER_COMPOSE_FILES_DIRECTORY}/gpu-docker*)
-          $GSUTIL_CMD cp gs://${INIT_BUCKET_NAME}/`basename ${GPU_DOCKER_COMPOSE}` $GPU_DOCKER_COMPOSE
-          COMPLETE_JUPYTER_DOCKER_COMPOSE="-f $JUPYTER_DOCKER_COMPOSE -f $GPU_DOCKER_COMPOSE"
-        fi
-
-        ${DOCKER_COMPOSE} ${COMPLETE_JUPYTER_DOCKER_COMPOSE} stop
```

Comment:
That was the culprit, the combination of docker stop and then remove did exactly that... Stopping the jupyter container and then removing it, which left no choice to docker compose up but to recreate the container, meaning that any user change applied outside of the mounted volume (aka persistent disk) was lost.

This should do the trick, I added the --no-recreate as an extra precaution